### PR TITLE
Cognito dataset final state

### PR DESCRIPTION
**Pull request by CedricProfessionnel**

### CODE REVIEW

File: cypress/integration/CognitoServer/basic.spec.js [hunk 1]
```diff
@@ -202,10 +160,31 @@ describe("Cognito Server Tests", () => {
     expect(dataset.name).to.include(samplesDummies.name)
   })
 
+  it("Remove Folder Samples", async () => {
+    await dm.removeSamplesFolder(samplesDummies.name)
+    var list = await dm.getListSamples(samplesDummies.name)
+    expect(list.length).to.equal(0)
+  })
+  it("Add asset ", async () => {
+    await dm.addAsset("chat.jpg", catPicture)
```
Comment (seveibar) [inline cypress/integration/CognitoServer/basic.spec.js:169]: cool, I think this API makes sense

File: src/CognitoDatasetManager.js [hunk 2]
```diff
@@ -289,9 +161,28 @@ class CognitoDatasetManager extends EventEmitter {
...
+      result.map(async (obj) => {
+        for (var i = 0; i < sampleIds.length; i++) {
+          if (obj.key.includes("/" + sampleIds[i] + ".json")) {
+            await Storage.remove(obj.key, {
+              level: this.dataPrivacyLevel,
+            })
+          }
+        }
+      })
+    )
```
Comment (seveibar) [inline src/CognitoDatasetManager.js:181]: this is potentially a lot of simultaneous htto requests

File: src/CognitoDatasetManager.js [hunk 3]
```diff
@@ -304,56 +195,378 @@ class CognitoDatasetManager extends EventEmitter {
...
+    var sampleName
+    if (!isEmpty(sample)) {
+      if (!isEmpty(this.getSampleUrl(sample))) {
+        sampleName = decodeURI(this.getSampleUrl(sample)).match(
+          `[^\\/\\\\&\\?]*\\.[a-zA-Z0-9]*(?:\\?|$)`
+        )
+      }
+    }
+    return sampleName
+  }
```
Comment (seveibar) [inline src/CognitoDatasetManager.js:540]: we'd want to move these into a base class or utility function

**Review by seveibar — APPROVED** [review]

**Review by seveibar — APPROVED** [review]

File: src/CognitoDatasetManager.js (hunk 3 above)
Comment (CedricProfessionnel) [reply src/CognitoDatasetManager.js:540]: If you talk about :
getSampleExtensionFromURL,
getSampleNameFromURL,
setSampleUrl
getSampleUrl
getSampleWithName

It's the idea

**Pull request merged**